riscv-semihosting/src/lib.rs

```diff
@@ -229,7 +231,8 @@ pub unsafe fn syscall1(_nr: usize, _arg: usize) -> usize {
                 .option pop
             ",
             inout("a0") nr,
-            in("a1") _arg,
+            inout("a1") arg,
```

Comment:
```suggestion
            inout("a1") arg => _,
```

The resulting value is not used, so currently it gets a warning.

https://github.com/rust-embedded/riscv/actions/runs/7388098852/job/20098197550

```
warning: value assigned to `arg` is never read
   --> riscv-semihosting/src/lib.rs:234:25
    |
234 |             inout("a1") arg,
    |                         ^^^
    |
    = help: maybe it is overwritten before being read?
    = note: `#[warn(unused_assignments)]` on by default
```